pkg/analytics/categories_test.go

```diff
@@ -0,0 +1,47 @@
+package analytics
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCategorizeCliArgs(t *testing.T) {
+	knownCommands := []string{"test", "code", "monitor", "iac", "rules", "describe", "sbom", "container"}
+	whitelistedFlags := []string{"unmanaged", "all-projects", "debug"}
+
+	testCases := []struct {
+		name           string
+		args           []string
+		expectedOutput []string
+		expectedError  error
+	}{
+		// Happy Paths
```

Comment:
Suggestion: The test cases are really helpful understanding the implementation! One thing I noticed was, that they are very similar in their pattern. The are actually many different orders and forms how to provide arguments, so that it is a bit tricky do do that correctly. For example take this command
```
snyk -d test ../folder/ --org=myorg --remote-repo-url https://github.com/snyk
```